### DISCUSSION TITLE
Auto-discovery of Cockpit hosts in local segment

### DISCUSSION THREAD

**Discussion by supakeen** [cockpit-project/cockpit#18472] [Q&A]
Hi, I have a number of machines running cockpit in my local network. Currently I have enabled a bunch of hosts but it's cumbersome to do this on 'all' machines that I might directly access cockpit from and want to use to switch to another machine.

How about a form of autodiscovery for cockpit hosts? mDNS or various other tools can be used to have cockpit respond to requests for 'who runs cockpit here'?

**Comment by martinpitt** [comment]
If you have many machines to manage, we actually don't recommend installing the webserver (i.e. what listens to port 9090) to all of them, and instead connect via SSH. You don't directly say what you do exactly, but it sounds like you are actually using the "connect to remote host" functionality, i.e. only have one "bastion host" with cockpit-ws?

mDNS etc. is difficult as there is literally nothing running on these remote machines. Even with the webserver, there's just cockpit.socket. Of course it *could* start another unit which actually "does" something, but it also goes against our ideal of "zero footprint".

I'm not totally averse to this idea of adding (opt-in) mDNS or so. But we actually plan to do this differently: In the not so far future, pretty much *every* computer will be able to give you at least a basic Cockpit UI, even if you don't have any cockpit-* packages installed.
